src/mds/Server.cc

```diff
@@ -442,6 +442,8 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
 
   if (!session) {
     dout(0) << " ignoring sessionless msg " << *m << dendl;
+    auto reply = make_message<MClientSession>(CEPH_SESSION_CLOSE);
+    mds->send_message(reply, m->get_connection());
```

Comment:
Better would be to send a reject message, see the code for handling blacklisted clients below in this method.

Reply:
You can also add the metadata for whwy the session is rejected. See:

`m->metadata["error_string"] = "sessionless";`
 
in the below code.